### PR TITLE
Deprecate static reflection

### PR DESCRIPTION
**Pull request by greg0ire**
It has become a burden to maintain, for no benefit since we cannot find
usages for it in Doctrine itself. The plan is to create a new major
version without this types, so that php 8 compatibility is achieved.

After this, the package will consist in only 2 classes that should
probably moved elsewhere.

### CODE REVIEW

File: docs/en/reference/index.rst [hunk 1]
```diff
@@ -1,85 +1,5 @@
...
-
-    echo $staticReflectionProperty->getName();
-
-    echo $staticReflectionProperty->getDeclaringClass();
-
-    echo $staticReflectionProperty->getDocComment();
-
-    print_r($staticReflectionProperty->getUseStatements());
+The main functionality of this package has been deprecated, you should
+not use it.
```
Comment (greg0ire) [inline docs/en/reference/index.rst:5]: Tell me what you think about this: does Doctrine usually document deprecated functionality? IMO it's best not to encourage people to build more on it.

**Comment by greg0ire** [comment]
https://github.com/doctrine/reflection/issues/24 added one TypedNoDefaultReflectionProperty of the 2 classes for the ORM, and is referenced in Persistence, maybe it should be moved to that lib?

**Comment by greg0ire** [comment]
`RuntimePublicReflectionProperty` is used in `persistence` and ORM, maybe it should be moved to `persistence`?

File: docs/en/reference/index.rst (hunk 1 above)
Comment (ostrolucky) [reply docs/en/reference/index.rst:5]: Wording is strange. So are we deprecating the package, or not? If we are, wording should say so. If we aren't, why is this saying to not use the package if there is still some use?

**Comment by ostrolucky** [comment]
I agree

File: docs/en/reference/index.rst (hunk 1 above)
Comment (morozov) [reply docs/en/reference/index.rst:5]: Yeah… if the main functionality of the package is deprecated, then what should it be used for and what's its new main functionality? Deprecating the whole thing and moving the relevant pieces elsewhere might be a more proper decision.

**Comment by malarzm** [comment]
ODM started using the package recently with https://github.com/doctrine/mongodb-odm/pull/2188

**Comment by greg0ire** [comment]
@malarzm do you remember why the dependency was added? Asking because I can't find a reference to the `Common\Reflection` namespace in that PR…

**Comment by malarzm** [comment]
@greg0ire the dependency was added to ensure we get latest `doctrine/reflection` including all the fixes for PHP 7.4 and uninitialized properties. I might have jumped the gun thinking the classes would be incorporated into ORM, `doctrine/persistence` would work just fine for ODM 👍

**Comment by greg0ire** [comment]
I made some progress regarding the move at https://github.com/doctrine/persistence/pull/122 , but I'm not quite there yet, help is welcome.

**Pull request closed without merging**